includes/discovery/sensors/dbm/fabos.inc.php

```diff
@@ -1,70 +1,45 @@
 <?php
 
+use LibreNMS\Util\Oid;
 use LibreNMS\Util\Rewrite;
 
-$fabosSfpRxPower = snmpwalk_array_num($device, '.1.3.6.1.4.1.1588.2.1.1.1.28.1.1.4'); // FA-EXT-MIB::swSfpRxPower
-$fabosSfpTxPower = snmpwalk_array_num($device, '.1.3.6.1.4.1.1588.2.1.1.1.28.1.1.5'); // FA-EXT-MIB::swSfpTxPower
-if (! empty($fabosSfpRxPower) || ! empty($fabosSfpTxPower)) {
-    $ifDescr = snmpwalk_group($device, 'ifDescr', 'IF-MIB', 0)['ifDescr'] ?? [];
-    $ifAdminStatus = snmpwalk_group($device, 'ifAdminStatus', 'IF-MIB', 0)['ifAdminStatus'] ?? [];
-}
+$fabosSfpRxPower = SnmpQuery::hideMib()->mibs(['FA-EXT-MIB'])->numericIndex()->walk('FA-EXT-MIB::swSfpRxPower')->valuesByIndex();
+$fabosSfpTxPower = SnmpQuery::hideMib()->mibs(['FA-EXT-MIB'])->numericIndex()->walk('FA-EXT-MIB::swSfpTxPower')->valuesByIndex();
 
-foreach ($fabosSfpRxPower as $oid => $entry) {
-    foreach ($entry as $index => $current) {
-        if (is_numeric($current)) {
-            $ifIndex = $index + 1073741823;
-            if ($ifAdminStatus[$ifIndex] == '1') {
-                discover_sensor(
-                    null,
-                    'dbm',
-                    $device,
-                    ".$oid.$index",
-                    'swSfpRxPower.' . $index,
-                    'brocade',
-                    Rewrite::shortenIfName($ifDescr[$ifIndex]) . ' RX',
-                    1,
-                    1,
-                    -5,
-                    null,
-                    null,
-                    0,
-                    $current,
-                    'snmp',
-                    $ifIndex,
-                    'ports',
-                    null,
-                    'Receive Power'
-                );
-            }
-        }
-    }
+$fabosSfpPower = array_merge_recursive($fabosSfpTxPower, $fabosSfpRxPower);
+
+if (! empty($fabosSfpPower)) {
+    $ifDescr = SnmpQuery::hideMib()->mibs(['IF-MIB'])->walk('IF-MIB::ifDescr')->table(1) ?? [];
```

Comment:
You probably shouldn't be walking these tables at all, but you should walk them together instead of separately like this.
Also, avoid use of hideMib(), it is better to be explicit.

Reply:
Would you prefer me to use a query with ports model ? I could do that but what if port module is not yet run, that will break the tests (sensors is not dependant on ports).
What would be your best move is in that case ?

Reply:
This is not related to this code specifically, just generic advice.